### DISCUSSION TITLE
Add a callback before applying changes to the files.

### DISCUSSION THREAD

**Discussion by antosha417** [olimorris/codecompanion.nvim#415] [Ideas]
What do you think about adding a callback that will be called before codecompanion would apply the chages to the file?

My use case is that I want to commit that file automatically before any changes are applied to it.

**Comment by olimorris** [comment]
Totally up for this. Perhaps if in the config we have something like:

```lua
tools = {
  ["editor"] = {
    callback = "strategies.chat.tools.editor",
    description = "Update a buffer with the LLM's response",
    opts = {
      handlers = {
        after = function() end,
      }
    }
  },
}
```

Would welcome this as a PR!
